src/main/kotlin/uk/gov/justice/digital/hmpps/externalmovementsapi/service/TapOccurrenceModifications.kt

```diff
@@ -12,31 +12,37 @@ import uk.gov.justice.digital.hmpps.externalmovementsapi.domain.referencedata.Ta
 import uk.gov.justice.digital.hmpps.externalmovementsapi.domain.referencedata.getByKey
 import uk.gov.justice.digital.hmpps.externalmovementsapi.domain.referencedata.of
 import uk.gov.justice.digital.hmpps.externalmovementsapi.exception.ConflictException
+import uk.gov.justice.digital.hmpps.externalmovementsapi.model.AuditedAction
 import uk.gov.justice.digital.hmpps.externalmovementsapi.model.actions.occurrence.CancelOccurrence
 import uk.gov.justice.digital.hmpps.externalmovementsapi.model.actions.occurrence.OccurrenceAction
 import uk.gov.justice.digital.hmpps.externalmovementsapi.model.actions.occurrence.RescheduleOccurrence
+import uk.gov.justice.digital.hmpps.externalmovementsapi.service.history.OccurrenceHistory
 import java.util.UUID
 
 @Service
 class TapOccurrenceModifications(
+  private val transactionTemplate: TransactionTemplate,
   private val taoRepository: TemporaryAbsenceOccurrenceRepository,
   private val referenceDataRepository: ReferenceDataRepository,
+  private val occurrenceHistory: OccurrenceHistory,
 ) {
-  @Transactional
-  fun apply(id: UUID, action: OccurrenceAction) {
+  fun apply(id: UUID, action: OccurrenceAction): AuditedAction {
     ExternalMovementContext.get().copy(reason = action.reason).set()
-    val occurrence = taoRepository.getOccurrence(id)
-    when (action) {
-      is RescheduleOccurrence -> occurrence.reschedule(action)
-      is CancelOccurrence -> {
-        if (occurrence.status.code !in listOf(SCHEDULED.name, CANCELLED.name)) {
-          throw ConflictException("Temporary absence not currently scheduled")
-        } else {
-          occurrence.cancel(action) { domain, code -> referenceDataRepository.getByKey(domain of code) }
+    transactionTemplate.executeWithoutResult {
+      val occurrence = taoRepository.getOccurrence(id)
+      when (action) {
+        is RescheduleOccurrence -> occurrence.reschedule(action)
+        is CancelOccurrence -> {
+          if (occurrence.status.code !in listOf(SCHEDULED.name, CANCELLED.name)) {
+            throw ConflictException("Temporary absence not currently scheduled")
+          } else {
+            occurrence.cancel(action) { domain, code -> referenceDataRepository.getByKey(domain of code) }
+          }
         }
-      }
 
-      else -> throw IllegalArgumentException("Action not supported")
+        else -> throw IllegalArgumentException("Action not supported")
+      }
     }
+    return occurrenceHistory.latestAction(id)
```

Comment:
Given this is now outside a transaction (I think) there exists a tiny chance two users could update at the same time and be given the wrong history response. Is there a way of matching to the expected history element?

If not or it's a bunch of code, fine to ignore. It's incredibly unlikely to happen and is only for an informational confirmation page!